src/mail-app/mail/model/ConversationListModel.ts

```diff
@@ -560,18 +581,27 @@ export class ConversationListModel implements MailSetListModel {
 		return this.entityClient.loadMultiple(MailTypeRef, listId, elements)
 	}
 
-	private readonly _items = memoizedWithHiddenArgument(
+	private readonly _items: () => readonly Mail[] = memoizedWithHiddenArgument(
 		() => this.listModel.state.items,
-		(conversations) => conversations.map((conversation) => this.getLatestMailForConversation(conversation).mail),
+		(conversations) => this.getDisplayedMailsOfConversations(conversations),
 	)
 
 	private _mails = memoizedWithHiddenArgument(
 		() => this.mailMap,
 		(mailMap) => Array.from(mailMap.values()).map(({ mail }) => mail),
 	)
 
+	private getDisplayedMailsOfConversations(conversations: readonly LoadedConversation[]): Mail[] {
+		return conversations.map((conversation) => this.getDisplayedMailForConversation(conversation)?.mail).filter((mail) => mail != null) as Mail[]
```

Comment:
I wouldn't recommend casting the whole array, it might be error prone. You could do `.filter(isNotNull)`